src/dftbp/dftbplus/mainio.F90

```diff
@@ -4014,60 +4023,75 @@ subroutine writeMdOut2(fd, tStress, tPeriodic, tBarostat, isLinResp, eField, tFi
-          write(fd, format2U) 'Gibbs free energy including KE', energy%EGibbsKin, 'H',&
-              & Hartree__eV * energy%EGibbsKin, 'eV'
-        end if
+      if (printForces) then
+        write(fd, "(A)") "Forces (au)"
+        do ii = 1, size(derivs, dim=2)
+          write(fd, "(3E24.8)") -derivs(:, ii)
+        end do
+      end if
+      if (hasStress) then
+        write(fd, "(A)") "Total stress (au)"
+        do ii = 1, 3
+          write(fd, "(3E24.8)") totalStress(:,ii)
+        end do
+      end if
+      write(fd, format2Ue) "Volume", cellVol, "au^3", Bohr__AA**3 * cellVol, "A^3"
+      write(fd, format2Ue) "Pressure", cellPressure, "au", cellPressure * au__pascal, "Pa"
+      if (abs(pressure) < 1.0e-16_dp) then
+        write(fd, format2U) "Gibbs free energy", energy%EGibbs, "H",&
+            & Hartree__eV * energy%EGibbs, "eV"
+        write(fd, format2U) "Gibbs free energy including KE", energy%EGibbsKin, "H",&
+            & Hartree__eV * energy%EGibbsKin, "eV"
       end if
     end if
+
     if (isLinResp) then
       if (energy%Eexcited /= 0.0_dp) then
         write(fd, format2U) "Excitation Energy", energy%Eexcited, "H",&
             & Hartree__eV * energy%Eexcited, "eV"
       end if
       if (allocated(energiesCasida)) then
         do ii = 1, size(energiesCasida)
-          write(strTmp,"('Excitation ',I0)")ii
+          write(strTmp, "('Excitation ',I0)") ii
           write(fd, format2U) trim(strTmp), energiesCasida(ii), "H",&
               & Hartree__eV * energiesCasida(ii), "eV"
         end do
       end if
     end if
-    write(fd, format2U) 'Potential Energy', energy%EMermin,'H', energy%EMermin * Hartree__eV, 'eV'
-    write(fd, format2U) 'MD Kinetic Energy', energy%Ekin, 'H', energy%Ekin * Hartree__eV, 'eV'
-    write(fd, format2U) 'Total MD Energy', energy%EMerminKin, 'H',&
-        & energy%EMerminKin * Hartree__eV, 'eV'
-    write(fd, format2U) 'MD Temperature', tempIon, 'au', tempIon / Boltzmann, 'K'
+
+    write(fd, format2U) "Potential Energy", energy%EMermin, "H", energy%EMermin * Hartree__eV, "eV"
+    write(fd, format2U) "MD Kinetic Energy", energy%Ekin, "H", energy%Ekin * Hartree__eV, "eV"
+    write(fd, format2U) "Total MD Energy", energy%EMerminKin, "H",&
+        & energy%EMerminKin * Hartree__eV, "eV"
+    write(fd, format2U) "MD Temperature", tempIon, "au", tempIon / Boltzmann, "K"
+
     if (allocated(eField)) then
       if (allocated(eField%EFieldStrength)) then
-        write(fd, format1U1e) 'External E field', eField%absEField, 'au',&
-            & eField%absEField * au__V_m, 'V/m'
+        write(fd, format1U1e) "External E field", eField%absEField, "au",&
+            & eField%absEField * au__V_m, "V/m"
       end if
     end if
-    if (tFixEf .and. tPrintMulliken) then
-      write(fd, "(A, F14.8)") 'Net charge: ', sum(q0(:, :, 1) - qOutput(:, :, 1))
+
+    if (fixEf .and. printMulliken) then
+      write(fd, "(A, F14.8)") "Net charge: ", sum(q0(:, :, 1) - qOutput(:, :, 1))
```

Comment:
The coverage report that there is no fixed Fermi test case reaching this line, so is it worth adding a short one?

Reply:
See comment above.